src/ResourceManager/Sql/Commands.Sql/ServerCommunicationLink/Services/AzureSqlServerCommunicationLinkAdapter.cs

```diff
@@ -0,0 +1,155 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.Commands.Sql.Database.Model;
+using Microsoft.Azure.Commands.Sql.Database.Services;
+using Microsoft.Azure.Commands.Sql.ServerCommunicationLink.Model;
+using Microsoft.Azure.Commands.Sql.Server.Adapter;
+using Microsoft.Azure.Commands.Sql.Services;
+using Microsoft.Azure.Common.Authentication.Models;
+using Microsoft.Azure.Management.Sql.Models;
+
+namespace Microsoft.Azure.Commands.Sql.ServerCommunicationLink.Services
+{
+    /// <summary>
+    /// Adapter for Server communication link operations
+    /// </summary>
+    public class AzureSqlServerCommunicationLinkAdapter
+    {
+        /// <summary>
+        /// Gets or sets the AzureEndpointsCommunicator which has all the needed management clients
+        /// </summary>
+        private AzureSqlServerCommunicationLinkCommunicator Communicator { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Azure profile
+        /// </summary>
+        public AzureContext Context { get; set; }
```

Comment:
Same q, why the separate fields for context and subscription - this data should already be in the client, why keep it separately?